libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionText.java

```diff
@@ -1,157 +1,79 @@
+        // Arrival instructions
+        BannerInstructions instructions = step.bannerInstructions().get(1);
+        createInstructionText(instructions);
+      } else {
+        // Normal instructions
+        BannerInstructions instructions = step.bannerInstructions().get(0);
+        createInstructionText(instructions);
       }
-      return textFields;
-    }
-
-    // Multiple Destinations
-    if (hasMultipleDestinations(step)) {
-      return formatMultipleStrings(step.destinations(), exitText);
-    }
-
-    // Multiple Names
-    if (hasMultipleNames(step)) {
-      return formatMultipleStrings(step.name(), exitText);
-    }
-
-    // Destination or Street Name
-    if (hasDestination(step)) {
-      textFields.primaryText = destination(step);
-      return textFields;
-    } else if (hasName(step)) {
-      textFields.primaryText = name(step);
-      return textFields;
-    }
-
-    // Fall back to instruction
-    if (hasInstruction(step)) {
-      textFields.primaryText = instruction(step);
-      return textFields;
-    }
-    return textFields;
-  }
-
-  private boolean hasRefs(LegStep upComingStep) {
-    return !TextUtils.isEmpty(upComingStep.ref());
-  }
-
-  private boolean isMotorway(LegStep step) {
-    if (step.intersections() == null || step.intersections().isEmpty()) {
-      return false;
     }
-    if (step.intersections().get(0).classes() == null || step.intersections().get(0).classes().isEmpty()) {
-      return false;
-    }
-    return step.intersections().get(0).classes().contains("motorway");
-  }
-
-
-  private String instruction(LegStep upComingStep) {
-    return upComingStep.maneuver().instruction();
-  }
-
-  private boolean hasInstruction(LegStep upComingStep) {
-    return upComingStep.maneuver() != null
-      && !TextUtils.isEmpty(upComingStep.maneuver().instruction());
-  }
-
-  private String name(LegStep upComingStep) {
-    return upComingStep.name().trim();
-  }
-
-  private boolean hasName(LegStep upComingStep) {
-    return !TextUtils.isEmpty(upComingStep.name());
   }
 
-  private String destination(LegStep upComingStep) {
-    return upComingStep.destinations().trim();
+  private boolean hasBannerInstructions(LegStep step) {
+    return step.bannerInstructions() != null && !step.bannerInstructions().isEmpty();
   }
 
-  private boolean hasDestination(LegStep upComingStep) {
-    return !TextUtils.isEmpty(upComingStep.destinations());
+  private boolean isArrivalManeuverType(LegStep step) {
+    return step.maneuver() != null
+      && !TextUtils.isEmpty(step.maneuver().type())
+      && step.maneuver().type().contains(STEP_MANEUVER_TYPE_ARRIVE);
   }
 
-  private boolean hasMultipleDestinations(LegStep upComingStep) {
-    return !TextUtils.isEmpty(upComingStep.destinations())
-      && StringAbbreviator.splitter(upComingStep.destinations()).length > 1;
+  private boolean shouldUseArrivalInstruction(LegStep step) {
+    return step.bannerInstructions().size() > 1
```

Comment:
Should this be `greater than` 1 or simply `!isEmpty()`? Will there always be a secondary instruction when arriving?